mobile-ui/src/main/java/org/sagebionetworks/research/mobile_ui/inject/ShowStepFragmentModule.java

```diff
@@ -71,7 +72,7 @@
 
     public interface ShowStepFragmentFactory {
         @NonNull
-        ShowStepFragmentBase create(@NonNull StepView stepView);
+        Fragment create(@NonNull StepView stepView);
```

Comment:
I don't think this is a good idea.

Reply:
These factories should be used by PerformTask and the results should all conform to ShowStep. 

For Tracking, if you're using the Task/Step model, either you need to create something new and not use PerformTask, or you can try PerformTask -> ShowTrackingStep -> TrackingFragment. Use ShowTrackingStep as a dumb container just to leverage existing PerformTask and ShowStep and load the TrackingFragment/ViewModel that actually does the work, then call back to ShowStep/PerformTask to leverage handling results.

Reply:
What if I made a ShowStepFragment interface and had everything conform to that?

Reply:
Yea, that makes sense